components/membership/registration-impl/src/main/kotlin/net/corda/membership/impl/registration/dynamic/handler/mgm/ApproveRegistrationHandler.kt

```diff
@@ -70,64 +79,83 @@ internal class ApproveRegistrationHandler(
         val approvedBy = state.mgm
         val approvedMember = state.registeringMember
         val registrationId = state.registrationId
-        val persistState = membershipPersistenceClient.setMemberAndRegistrationRequestAsApproved(
-            viewOwningIdentity = approvedBy.toCorda(),
-            approvedMember = approvedMember.toCorda(),
-            registrationRequestId = registrationId,
-        )
-        val memberInfo = persistState.getOrThrow()
+        val messages = try {
+            val mgm = memberTypeChecker.getMgmMemberInfo(approvedBy.toCorda())
+                ?: throw CordaRuntimeException("Could not approve $registrationId - member ${approvedBy.x500Name} is an MGM.")
+            if (memberTypeChecker.isMgm(approvedMember)) {
+                throw CordaRuntimeException("Could not approve $registrationId by ${approvedMember.x500Name} it is not an MGM.")
```

Comment:
This might explain the situation better -
```suggestion
                throw CordaRuntimeException("The registration request: '$registrationId' cannot be approved by ${approvedMember.x500Name} as it is not an MGM.")
```